src/Shell/Task/ModelTask.php

```diff
@@ -268,6 +275,34 @@ public function findBelongsTo($model, array $associations)
     }
 
     /**
+     * find the table, if any, actually referenced by the passed key field.
+     *   Search tables in db for keyField; if found search key constraints
+     *   for the table to which it refers.
+     *
+     * @param null $keyField
+     * @return null
+     */
+    public function findTableReferencedBy($keyField = null){
+        $db = ConnectionManager::get($this->connection);
+        $schema = $db->schemaCollection();
+        $tables = $schema->listTables();                                // get tables in db
+
+        foreach ($tables as $table){
+            $meta = $schema->describe($table, ['forceRefresh'=>true]);  // get table meta data
+            $columns = $meta->columns();
+            if (!in_array($keyField, $columns)) continue;               // not in this table
+            $constraints = $meta->constraints();                        // found it; get constraints
+            foreach ($constraints as $constraint){
+                $constraint_info = $meta->constraint($constraint);      // get details of constraint
+                if(in_array($keyField, $constraint_info['columns'])) {  // match?
```

Comment:
Inline comments go one the line above. Most of these comments are not necessary though as the code is pretty straightforward.